docs/versioned_docs/version-7.7.x/configuration/overview.md

```diff
@@ -261,7 +261,7 @@ Provider specific options can be found on their respective subpages.
 
 `oauth2-proxy` supports having multiple upstreams, and has the option to pass requests on to HTTP(S) servers, unix socket or serve static files from the file system.
 
-HTTP and HTTPS upstreams are configured by providing a URL such as `http://127.0.0.1:8080/` for the upstream parameter. . This will forward all authenticated requests to the upstream server. If you instead provide `http://127.0.0.1:8080/some/path/` then it will only be requests that start with `/some/path/` which are forwarded to the upstream.
+HTTP and HTTPS upstreams are configured by providing a URL value. The path of the URL will be used to configure a matcher that incoming requests will be compared against, to decide which of the "upstream" servers to proxy the request to. A URL such as `http://127.0.0.1:8080/` for the upstream parameter will send all authenticated requests to 127.0.0.1:8080 . A path of the URL like `/foo` will match exactly and only `/foo` and not `/food` or `/foo/bar`, but a trailing slash is special and will be used as a prefix matcher; request URLs having that as a prefix will match it; for example `/foo/` will match an incoming `/foo/bar`. Since you may configure many values of `--upstream`, the longest length (and therefore most specific) configured matcher string will have priority over those that match more broadly.
```

Comment:
@chadmiller-saq 
How about rephrasing it like this?

```suggestion
To configure HTTP/HTTPS upstreams, you have to provide a URL. The URL's path determines how incoming requests are matched to specific upstream servers:

* **Exact Match:** Paths without a trailing slash match exactly. For example, /foo matches only /foo and not /food or /foo/bar.
* **Prefix Match:** Paths with a trailing slash are used as prefixes. For example, /foo/ matches incoming requests that start with /foo/, such as /foo/bar.
Matching Priority: If multiple upstreams match an incoming request, the upstream with the longest matching path (most specific) takes priority.
```

Will render like:

To configure HTTP/HTTPS upstreams, you have to provide a URL. The URL's path determines how incoming requests are matched to specific upstream servers:

* **Exact Match:** Paths without a trailing slash match exactly. For example, /foo matches only /foo and not /food or /foo/bar.
* **Prefix Match:** Paths with a trailing slash are used as prefixes. For example, /foo/ matches incoming requests that start with /foo/, such as /foo/bar.
Matching Priority: If multiple upstreams match an incoming request, the upstream with the longest matching path (most specific) takes priority.

Reply:
If you are fine with this change? Please apply it to the file `docs/docs/configuration/overview.md` as well :)

Reply:
Thanks! I incorporated your excellent formatting suggestions. It was a dense paragraph!

Reply:
No worries :) Thanks for keeping an eye out for good documentation. Much appreciated and often overlooked!